File: VB/ColorMappingExample/PriorityToColorConverter.vb
Imports System
Imports System.Globalization
Imports System.Windows.Data
Imports System.Windows.Markup
Imports System.Windows.Media

Namespace ColorMappingExample

    Friend Class PriorityToColorConverter
        Inherits MarkupExtension
        Implements IValueConverter

        Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.Convert
            If TypeOf value Is Priority Then
                Dim priorityValue = CType(value, Priority)
                Select Case priorityValue
                    Case Priority.High
                        Return Colors.Red
                    Case Priority.Normal
                        Return Colors.Yellow
                    Case Priority.Low
                        Return Colors.Green
                    Case Else
                        Return Colors.Black
                End Select
            End If

            Return Colors.Transparent
        End Function

        Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            Throw New NotImplementedException()
        End Function

        Public Overrides Function ProvideValue(ByVal serviceProvider As IServiceProvider) As Object
            Return Me
        End Function
    End Class
End Namespace

File: CS/ColorMappingExample/PriorityToColorConverter.cs
using System;
using System.Globalization;
using System.Windows.Data;
using System.Windows.Markup;
using System.Windows.Media;

namespace ColorMappingExample {
    class PriorityToColorConverter : MarkupExtension, IValueConverter {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture) {
            if (value is Priority val) {
                switch (val) {
                    case Priority.High:
                    return Colors.Red;

                    case Priority.Normal:
                    return Colors.Yellow;

                    case Priority.Low:
                    return Colors.Green;

                    default:
                    return Colors.Black;
                }
            }
            return Colors.Transparent;
        }
        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture) {
            throw new NotImplementedException();
        }
        public override object ProvideValue(IServiceProvider serviceProvider) {
            return this;
        }
    }
}
